lib/block-supports/elements.php

```diff
@@ -38,7 +38,7 @@ function gutenberg_render_elements_support( $block_content, $block ) {
 	}
 	$link_color_declaration = esc_html( safecss_filter_attr( "color: $link_color" ) );
 
-	$style = "<style>.$class_name a{" . $link_color_declaration . " !important;}</style>\n";
```

Comment:
This is going to need changes in the test. Will update them once we're certain this is the right direction.

Reply:
This is going to have 011 specificity.

I've looked at all current cases of core blocks and this change will still work well with global styles. The reason is that the current global styles generated for links have 011 or 002, depending on whether they are `.wp-block-* a` or `element a`. The element styles are loaded after the global styles, so the element styles win.

I was thinking that we could add the `:not(.has-link-color)` selector to the global styles we generate. Should any block have higher than 011 specificity this selector will prevent it from applying. However! This is also not complete: it won't work for some blocks that declare `__experimentalSelector` and target an inner element. Use case:

- some block has a wrapper such as `.wp-block-wrapper`
- for whatever reason, the block declares that the styles should match some inner element instead `__experimentalSelector: `.wp-block-wrapper-inner .wp-block-wrapper-extra`

Adding `:not(.has-link-color)` in this situation would render `.wp-block-wrapper-inner .wp-block-wrapper-extra:not(.has-link-color) { /* link color */ }` which has higher specificity that the link color element and still targets the links in the block because the `.has-link-color` class has been applied to the wrapper.